platform.txt

```diff
@@ -69,11 +69,20 @@ compiler.arm.cmsis.ldflags="-L{runtime.tools.CMSIS-5.5.1.path}/CMSIS/DSP/Lib/GCC
 
 # USB Flags
 # ---------
-build.usb_flags=-DUSBCON {build.usb_speed} -DUSBD_VID={build.vid} '-DUSB_MANUFACTURER={build.usb_manufacturer}' '-DUSB_PRODUCT="{build.board}"' -DHAL_PCD_MODULE_ENABLED 
-
-# Default usb manufacturer will be replaced at compile time using
-# numeric vendor ID if available or by board's specific value.
-build.usb_manufacturer="Unknown"
+build.usb_flags=-DUSBCON {build.usb_speed} -DUSBD_VID={build.vid} -DUSBD_PID={build.pid} -DHAL_PCD_MODULE_ENABLED
```

Comment:
As '-DUSB_PRODUCT="{build.board}"' is removed this could not build as it is used 

```
#ifdef USBD_USE_HID_COMPOSITE
#define USBD_CLASS_CONFIGURATION_HS_STRING  CONCATS(USB_PRODUCT, "HID Config")
#define USBD_CLASS_INTERFACE_HS_STRING      CONCATS(USB_PRODUCT, "HID Interface")
#define USBD_CLASS_CONFIGURATION_FS_STRING  CONCATS(USB_PRODUCT, "HID Config")
#define USBD_CLASS_INTERFACE_FS_STRING      CONCATS(USB_PRODUCT, "HID Interface")
#endif /* USBD_USE_HID_COMPOSITE */
#ifdef USBD_USE_CDC
#define USBD_CLASS_CONFIGURATION_HS_STRING  CONCATS(USB_PRODUCT, "CDC Config")
#define USBD_CLASS_INTERFACE_HS_STRING      CONCATS(USB_PRODUCT, "CDC Interface")
#define USBD_CLASS_CONFIGURATION_FS_STRING  CONCATS(USB_PRODUCT, "CDC Config")
#define USBD_CLASS_INTERFACE_FS_STRING      CONCATS(USB_PRODUCT, "CDC Interface")
#endif /* USBD_USE_CDC */
```

Reply:
Ah, I changed that to `BOARD_NAME` for the mfg and product string, but forgot to do this here.

Reply:
@matthijskooijman 
In fact, you remove it but it should be better to keep this define: `USB_PRODUCT`
By default it is set to `BOARD_NAME` but it could be redefined.
It is the same in Arduino core.

Reply:
I thought it better to remove it, since otherwise then you have both `USB_PRODUCT_STRING`, which defines the full product string, and `USB_PRODUCT` which is used as a component to build the default strings. And since `USP_PRODUCT` was unconditionally set to `{build.board}` anyway, it seemed easier to just use `BOARD_NAME` in the code directly. My thought was that if someone wants to define a custom product name, they can just define `USB_PRODUCT_STRING` and set the complete value, and if they do not care, then just using a default value based on the `BOARD_NAME` is fine.

Looking at the Arduino core, things are a bit different anyway, since there they just use `USB_MANUFACTURER` and `USB_PRODUCT` to define the full strings and just use `"USB IO Board"` as the default for `USB_PRODUCT` if not defined (so there is no class indication or board name in the product name).

Reply:
OK fine for me.
Thanks